### PR TITLE
Eng 2104 [1/2][backend] fix a bug where mongo collection preview crashes

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR introduce backend fixes for mongo data preview crash. The root cause is our current `to_json` data frame serialization is not compatible with Mongo's python types, which relies on `bson`. This randomly cause json serialization to fail.

We introduced a new serialization type `bson_table` to cover this case. For now, the logic is -
* df extracted from mongo is with `bson_table` serialization type.
* operator with any `bson_table` inputs will have `bson_table` to serialize df outputs
It might work by introducing a new artf type, but the current solution allow us to fix the issue with narrower range of change.

I noticed some `enum` migration are not finished in the middle, so there are some package import change mixed up in this PR.

## Related issue number (if any)
ENG-2104

## Loom demo (if any)
WIP

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: sdk/aqueduct/utils/serialization.py [hunk 1]
```diff
@@ -147,10 +158,17 @@ def serialize_val(val: Any, serialization_type: SerializationType) -> bytes:
 
 
 def artifact_type_to_serialization_type(
-    artifact_type: ArtifactType, content: Any
+    artifact_type: ArtifactType,
+    # derived_from_bson specifies if the artifact is derived from a bson object
+    # and thus requires bson encoding.
+    # For now, it only applies to data frames extracted / transformed from Mongo.
```
Comment (cw75) [inline sdk/aqueduct/utils/serialization.py:164]: "DataFrames"

File: sdk/aqueduct/utils/utils.py [hunk 2]
```diff
@@ -105,7 +105,12 @@ def parse_user_supplied_id(id: Union[str, uuid.UUID]) -> str:
 
 
 def construct_param_spec(val: Any, artifact_type: ArtifactType) -> ParamSpec:
-    serialization_type = artifact_type_to_serialization_type(artifact_type, val)
+    serialization_type = artifact_type_to_serialization_type(
+        artifact_type,
+        # not derived from bson. This is irrelevant for now as we don't support table parameters.
```
Comment (cw75) [inline sdk/aqueduct/utils/utils.py:110]: We support parameters of any type, so I do think we support table parameters; we just don't expect the table parameter to be serialized in the bson form, so I think this change is still fine. If so we just need to update this comment to be more accurate.

**Review by cw75 — APPROVED** [review]
@likawind LGTM! Left a couple nits inline. I do think creating a separate artifact type for mongo table is cleaner, as it removes the need for the `derived_from_bson` argument which is not used for other artifact types. So if this fix works for the user, I think we should make an effort to clean this up. It will involve a data migration effort after merging this PR though.

**Comment by likawind** [comment]
@cw75 I agree using a new artifact type could make a few codes cleaner here. This is the task: https://linear.app/aqueducthq/issue/ENG-2153/consider-making-mongo-artifacts-a-new-artf-type-instead-of-a-new

**Pull request merged**